monitoring/custom-metrics/custom-metrics.py

```diff
@@ -0,0 +1,337 @@
+                                # specific collection in a specific database
+                                if collection in collections_by_database[database_to_monitor]:
+                                    namespaces_to_monitor = monitor_namespace(database_to_monitor, collection, namespaces_to_monitor)
+
+            for namespace in namespaces_to_monitor:
+                tokens = namespace.split(".")
+                database_name = tokens[0]
+                collection_name = tokens[1]
+                database = DATABASE_CLIENT[database_name]
+                collection_statistics = database.command("collStats", collection_name)
+                if parameters["log_collection_size"] is True:
+                    log_collection_size_metric(parameters["cluster_name"], database_name, collection_name, collection_statistics["storageSize"])
+
+                if parameters["log_collection_index_count"] is True:
+                    log_index_count_metric(parameters["cluster_name"], database_name, collection_name, collection_statistics["nindexes"])
+
+                if parameters["log_collection_index_size"] is True:
+                    for index_name in collection_statistics["indexSizes"]:
+                        log_index_size_metric(parameters["cluster_name"], database_name, collection_name, index_name, collection_statistics["indexSizes"][index_name])
+
+def main():
+    """custom_metrics script entry point."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--cluster_name',
+                        required=True,
+                        type=str,
+                        help='Name of cluster for Amazon CloudWatch custom metric')
+
+    parser.add_argument('--uri',
+                        required=True,
+                        type=str,
+                        help='Amazon DocumentDB Connection URI')
+
+    parser.add_argument('--namespaces',
+                        required=True,
+                        type=str,
+                        help="comma separated list of namespaces to monitor")
+
+    parser.add_argument('--collection_count',
+                        action='store_true',
+                        help="log cluster collection count")
+
+    parser.add_argument('--database_count',
+                        action='store_true',
+                        help="log cluster database count")
+
+    parser.add_argument('--user_count',
+                        action='store_true',
+                        help="log cluster user count")
+
+    parser.add_argument('--collection_size',
+                        action='store_true',
+                        help="log collection size")
+
+    parser.add_argument('--index_count',
+                        action='store_true',
+                        help="log collection index count")
+
+    parser.add_argument('--index_size',
+                        action='store_true',
+                        help="log collection index size")
+
+    args = parser.parse_args()
```

Comment:
Add Python 3.7 minimum check with optional opt-out as in fl-multiprocess.py and other scripts in this repo.

Reply:
Added Python version check